File: frontend/src/api/studio.ts
import { getJson } from "./client";

export type WorkspaceSummary = {
  documents: number;
  chunks: number;
  graph_nodes: number;
  graph_edges: number;
};

export type EvaluationSummary = {
  available: boolean;
  run_id: string | null;
  timestamp: string | null;
  dataset_version: string | null;
  total_cases: number;
  passed_cases: number;
  failed_cases: number;
  pass_rate: number | null;
  intent_accuracy: number | null;
  route_accuracy: number | null;
  citation_validity_rate: number | null;
  grounded_answer_pass_rate: number | null;
  refusal_accuracy: number | null;
  fabricated_answer_rate: number | null;
  core_total: number;
  core_passed: number;
  holdout_total: number;
  holdout_passed: number;
};

export function getWorkspaceSummary(): Promise<WorkspaceSummary> {
  return getJson<WorkspaceSummary>("/api/v1/workspace/summary");
}

export function getEvaluationSummary(): Promise<EvaluationSummary> {
  return getJson<EvaluationSummary>("/api/v1/evaluation/summary");
}

File: backend/app/schemas/studio.py
from __future__ import annotations

from datetime import datetime

from pydantic import Field

from backend.app.schemas.documents import StrictBaseModel


class WorkspaceSummaryResponse(StrictBaseModel):
    documents: int = Field(ge=0)
    chunks: int = Field(ge=0)
    graph_nodes: int = Field(ge=0)
    graph_edges: int = Field(ge=0)


class EvaluationSummaryResponse(StrictBaseModel):
    available: bool
    run_id: str | None = None
    timestamp: datetime | None = None
    dataset_version: str | None = None
    total_cases: int = Field(default=0, ge=0)
    passed_cases: int = Field(default=0, ge=0)
    failed_cases: int = Field(default=0, ge=0)
    pass_rate: float | None = Field(default=None, ge=0.0, le=1.0)
    intent_accuracy: float | None = Field(default=None, ge=0.0, le=1.0)
    route_accuracy: float | None = Field(default=None, ge=0.0, le=1.0)
    citation_validity_rate: float | None = Field(default=None, ge=0.0, le=1.0)
    grounded_answer_pass_rate: float | None = Field(default=None, ge=0.0, le=1.0)
    refusal_accuracy: float | None = Field(default=None, ge=0.0, le=1.0)
    fabricated_answer_rate: float | None = Field(default=None, ge=0.0, le=1.0)
    core_total: int = Field(default=0, ge=0)
    core_passed: int = Field(default=0, ge=0)
    holdout_total: int = Field(default=0, ge=0)
    holdout_passed: int = Field(default=0, ge=0)
